### PR TITLE
Permit hotkey behavior within checkboxes and radio buttons

### LINKED ISSUE #145: Hotkeys get disabled when focusing form controls
### Bug Report

- __Package version(s)__: 1.0.1
- __Browser and OS versions__: macOS, Chrome 54

#### Describe the issue:

Global hotkeys are correctly disabled when using a **text** input, because the user is looking to input text.

They should still be enabled after focusing checkboxes, switches, radios. It's not the case now, see GIF below.

#### Actual behavior:

screen recording google chrome

#### Expected behavior:

Focusing an `<input>` shouldn't kill hotkeys, unless the input is a text input, I suppose?

### PR DESCRIPTION
**Pull request by cmslewis**
#### PR checklist

- [x] Addresses an existing issue: fixes #145 
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests

#### What changes did you make?

Hotkey behavior is now properly triggered when the user is focused within a checkbox or radio button. (Previously, the hotkey behavior was prevented unnecessarily.) Also added new unit tests to verify this behavior.

#### Is there anything you'd like reviewers to focus on?

Nothing in particular.

### CODE REVIEW

File: packages/core/test/hotkeys/hotkeysTests.tsx [hunk 1]
```diff
@@ -92,6 +101,54 @@ describe("Hotkeys", () => {
             expect(globalHotkeySpy.called).to.be.false;
         });
 
+        it("ignores hotkeys when inside number input", () => {
```
Comment (giladgray) [inline packages/core/test/hotkeys/hotkeysTests.tsx]: this is an ideal use case for a little test helper.

```ts
assertInputAllowsKeys(type: string, allowsKeys: boolean) {
  // fill me out please
}

assertInputAllowsKeys("number", false);
assertInputAllowsKeys("checkbox", true);
// ...
```

Comment (cmslewis) [inline packages/core/test/hotkeys/hotkeysTests.tsx]: Sure. I <3 little test helpers.

File: packages/core/test/hotkeys/hotkeysTests.tsx [hunk 2]
```diff
@@ -53,6 +53,21 @@ describe("Hotkeys", () => {
             }
         }
 
+        function assertInputAllowsKeys(type: string, allowsKeys: boolean) {
```
Comment (adidahiya) [inline packages/core/test/hotkeys/hotkeysTests.tsx]: I haven't looked at these test files recently, but I usually like to put helper functions last in the file / namespace scope / test suite. IMO this promotes file readability by letting you see the important things first -- the higher-level code that references these helper functions.

Comment (cmslewis) [inline packages/core/test/hotkeys/hotkeysTests.tsx]: Done

**Pull request merged**